### PR TITLE
Maintain control_deps in replace_node

### PR DESCRIPTION
**Pull request by diyessi**
Add helpers for mainting control_deps in other ways
Fix compiler warning about shadowing a local

### CODE REVIEW

File: src/ngraph/node.cpp [hunk 1]
```diff
@@ -334,9 +334,54 @@ const std::set<std::shared_ptr<Node>>& Node::get_control_dependencies() const
...
+    return m_control_dependents;
+}
+
 void Node::add_control_dependency(std::shared_ptr<Node> node)
 {
     m_control_dependencies.insert(node);
+    node->m_control_dependents.insert(this);
+}
+
+void Node::add_node_control_dependencies(std::shared_ptr<Node> source_node)
```
Comment (aprocter) [inline src/ngraph/node.cpp]: I think it'd be better if this and `add_node_control_dependents` were free functions.

File: src/ngraph/node.hpp [hunk 2]
```diff
@@ -238,16 +238,26 @@ namespace ngraph
...
+        /// Remove the dependency of this node on node
+        void remove_control_dependency(std::shared_ptr<Node> node);
+
+        /// Remove all dependencies from this node
+        void clear_control_dependencies();
+
+        /// Remove this node as a dependency from all dependent nodes
+        void clear_control_dependents();
+
+        /// This node aborbs the control dependencies of source_node
```
Comment (indie) [inline src/ngraph/node.hpp]: absorbs? aborts?

File: src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp [hunk 3]
```diff
@@ -313,9 +313,6 @@ void ngraph::runtime::gpu::pass::LSTMFusion::construct_lstm_fprop()
         auto ht_output = std::make_shared<op::GetOutputElement>(lstm, 0);
         auto ct_output = std::make_shared<op::GetOutputElement>(lstm, 2);
 
-        NGRAPH_CHECK(lstm->get_outputs().at(0).get_inputs().size() == 2,
-                     "Lstm node doesnt have two outputs");
-
         // Now identify the nodes which consumes the output of LSTM nodes
```
Comment (indie) [inline src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp]: identify the nodes which consume

File: src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp [hunk 4]
```diff
@@ -313,9 +313,6 @@ void ngraph::runtime::gpu::pass::LSTMFusion::construct_lstm_fprop()
...
         // and replace them accordingly
         // find the user's for {ht|ct} and replace them with lstm_goe_1
```
Comment (indie) [inline src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp]: Actually might want to reword the whole comment; "find the user's" makes this read like it needs an object.

File: src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp [hunk 5]
```diff
@@ -313,9 +313,6 @@ void ngraph::runtime::gpu::pass::LSTMFusion::construct_lstm_fprop()
         auto ht_output = std::make_shared<op::GetOutputElement>(lstm, 0);
         auto ct_output = std::make_shared<op::GetOutputElement>(lstm, 2);
 
-        NGRAPH_CHECK(lstm->get_outputs().at(0).get_inputs().size() == 2,
-                     "Lstm node doesnt have two outputs");
```
Comment (indie) [inline src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp]: Maybe something like "LSTM node lacks two outputs"

File: src/ngraph/node.hpp (hunk 2 above)
Comment (diyessi) [reply src/ngraph/node.hpp]: absorbs

File: src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp (hunk 4 above)
Comment (diyessi) [reply src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp]: The comment is being deleted.

File: src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp (hunk 5 above)
Comment (diyessi) [reply src/ngraph/runtime/gpu/pass/gpu_rnn_fusion.cpp]: deleted section

**Pull request merged**